src/ucp/core/ucp_request.inl

```diff
@@ -693,9 +693,14 @@ static UCS_F_ALWAYS_INLINE void
 ucp_recv_desc_release(ucp_recv_desc_t *rdesc)
 {
     void *uct_desc;
+    uint16_t flags = rdesc->flags;
+
+    /* reset flags to determine if the descriptor will be wrongly used
+     * after release */
+    rdesc->flags = 0;
```

Comment:
let's do it for AM RTS handling only, to avoid performance impact for other flows (tag, stream, am eager, etc)